WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductCoordinator.swift

```diff
@@ -279,14 +278,15 @@ private extension AddProductCoordinator {
     }
 
     func startProductCreationWithAI() {
-        let coordinator = AddProductWithAICoordinator(siteID: siteID,
-                                                      source: source,
-                                                      sourceNavigationController: navigationController,
-                                                      onProductCreated: { [weak self] product in
-            self?.onProductCreated(product)
-        })
-        self.addProductWithAICoordinator = coordinator
-        coordinator.start()
+        let viewController = AddProductWithAIContainerHostingController(viewModel: .init(siteID: siteID,
+                                                                                         source: source,
+                                                                                         onCancel: { [weak self] in
+            self?.navigationController.dismiss(animated: true)
+        },
+                                                                                         onCompletion: {
+            // TODO: Product saved
```

Comment:
👀 Please trigger `onProductCreated` here to avoid missing it later when we handle the completion.

```suggestion
            // TODO: Product saved
            self?.onProductCreated(product)
```

Reply:
I missed that. Fixed in https://github.com/woocommerce/woocommerce-ios/pull/10738/commits/dfd0a43efee254d994023ff9f089a09b41b207b1